features/nanostack/FEATURE_NANOSTACK/mbed-mesh-api/source/ThreadInterface.cpp

```diff
@@ -39,7 +39,12 @@ int ThreadInterface::connect()
     // Release mutex before blocking
     nanostack_unlock();
 
+#if (true == MBED_CONF_MBED_MESH_API_THREAD_USE_STATIC_LINK_CONFIG)
     int32_t count = connect_semaphore.wait(30000);
+#else
+    // static link configuration is not used, wait for commissioning to happen
+    int32_t count = connect_semaphore.wait(osWaitForever);
+#endif
```

Comment:
@SeppoTakalo @kjbracey-arm I have issue with the way interface is working differently now.

Thread is meant to be fire and forget. it will get the internet after weeks or by attaching to existing network in couple seconds or forming own network.

when End devices don't find parent they will not get the network in 30 seconds, but will get when some router becomes available --> should wait forever

Should we have method to identify when we have local connectivity or global connectivity.
And this should delay only to the point when we have configuration and stack is initialised correctly